### ISSUE TITLE
Failure to expand `matches!`

### ISSUE THREAD

**Issue by PizzaCrust** [intellij-rust/intellij-rust#7322]
## Environment

* **IntelliJ Rust plugin version:** 0.4.149.3912-211-nightly
* **Rust toolchain version:** 1.54.0-nightly (35fff69d0 2021-06-06) x86_64-pc-windows-msvc
* **IDE name and version:** IntelliJ IDEA 2021.1.2 Community Edition (IC-211.7442.40)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
Unable to expand the macro invocation, `matches!`, even though it's valid
(I believe this used to expand correctly?)

![image](https://user-images.githubusercontent.com/15954148/121215074-e14ed400-c8a9-11eb-90dd-8368aba74b79.png)

Potentially relevant logs: 
[idea.log](https://github.com/intellij-rust/intellij-rust/files/6617954/idea.log)

## Steps to reproduce
```rust
#[cfg(test)]
mod tests {
    #[test]
    fn it_works() {
        println!("{}", matches!(1, 1))
    }
}
```

**Comment by PizzaCrust** [comment]
It seems to work on stable but it doesn't work on nightly for some reason

**Comment by vlad20012** [comment]
Fixed by #7373
